### ISSUE TITLE
Rewrite "osd crush location" to "crush location" during jewel->luminous upgrade

### ISSUE THREAD

**Issue by smithfarm** [SUSE/DeepSea#761]
There is a potential problem when automatically upgrading a Jewel cluster to Luminous, if the ceph.conf of any storage node where the CRUSH location has been manually customized via an "osd crush location" setting. If this setting is left as-is, after the upgrade the OSDs will move from the customized location back to the default one, which can have, shall we say, "unwanted consequences". [1]

Would it make sense for DeepSea's upgrade automation to rewrite these settings by doing something like `sed -i -e 's/^osd crush location/crush location/ /etc/ceph/ceph.conf` as it upgrades each node?

[1] Sage is adding the following to the Luminous release notes:

* The ``osd crush location`` config option is no longer supported.  Please   update your ceph.conf to use the ``crush location`` option instead.  Be sure   to update your config file to avoid any movement of OSDs from your customized   location back to the default one.

**Comment by smithfarm** [comment]
@jschmid1 @kmroz @tserong What do you think?

**Comment by tserong** [comment]
> Would it make sense for DeepSea's upgrade automation to rewrite these settings by doing something like sed -i -e 's/^osd crush location/crush location/' /etc/ceph/ceph.conf as it upgrades each node?

Yeah, probably. At least, we don't want OSDs suddenly moving around. But it might mean sedding the relevant bits of /srv/salt/ceph/configuration/files/*conf*, rather than sedding /etc/ceph/ceph.conf on each node...

**Comment by smithfarm** [comment]
> But it might mean sedding the relevant bits of /srv/salt/ceph/configuration/files/conf, rather than sedding /etc/ceph/ceph.conf on each node...

Yes . . . and the sedding could be done as DeepSea converts the latter (`/etc/ceph/ceph.conf`) into the former (`/srv/salt/ceph/configuration/files/conf`) during the upgrade?

**Comment by jschmid1** [comment]
I'd say this is a rather critical option and should be touched manually. Opened a doc bug for it.

**Comment by smithfarm** [comment]
@jschmid1 Would it make sense for the ceph.maintenance.upgrade orchestration to implement a sanity check for this, and abort if has not been touched manually?

**Comment by jschmid1** [comment]
yes, that sounds good. 

Assigning self to track task
